framework/src/utils/LinearInterpolation.C

```diff
@@ -70,10 +87,22 @@ T
 LinearInterpolationTempl<T>::sampleDerivative(const T & x) const
 {
   // endpoint cases
-  if (x < _x[0])
-    return 0.0;
-  if (x >= _x[_x.size() - 1])
-    return 0.0;
+  if (_extrap)
+  {
+    if (x <= _x[0])
+      return (_y[1] - _y[0]) / (_x[1] - _x[0]);
+    if (x >= _x.back())
+    {
+      return (_y[_y.size() - 2] - _y.back()) / (_x[_x.size() - 2] - _x.back());
+    }
```

Comment:
no braces around single line statements

Reply:
> The style issue can be addressed later if you need this PR now

I am afraid I might forget later so I am pushing a new PR to fix it now.